### ISSUE TITLE
[SWC] Module parse failed: The keyword 'yield' is reserved

### ISSUE THREAD

**Issue by felixmosh** [vercel/next.js#31882]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome 88

### What operating system are you using?

macOS

### How are you deploying your application?

other

### Describe the Bug

Code with `for` loop with `await` inside throws error.

### Expected Behavior

It should work, it is a valid JS

### To Reproduce

Create a page like:
```js
import { useEffect } from 'react';

async function injectExternalScripts() {
  const list = ['https://external.com/x.js', 'https://external.com/y.js'];

  for (const src of list) {
    await new Promise((resolve, reject) => {
      const s = document.createElement('script');
      s.src = src;
      s.onload = resolve;
      s.onerror = reject;
      document.body.appendChild(s);
    });
  }
}

export default function Home() {
  useEffect( async() => {
    await injectExternalScripts();
  },[])
  return (
    <div>
      <p>Hello World</p>
    </div>
  )
}
```

Run `next build`

**Comment by iicdii** [comment]
Same error happens to me.

```shell
[dev:frontend] wait  - compiling /_error...
[dev:frontend] error - ./utils/excel/import-excel.js
[dev:frontend] Module parse failed: The keyword 'yield' is reserved (202:84)
[dev:frontend] File was processed with these loaders:
[dev:frontend]  * ./node_modules/@next/react-refresh-utils/loader.js
[dev:frontend]  * ./node_modules/next/dist/build/webpack/loaders/next-swc-loader.js
[dev:frontend] You may need an additional loader to handle the result of these loaders.
[dev:frontend] |                                     case 'product':
[dev:frontend] |                                         var ref;
[dev:frontend] >                                         query = value && row.type === 'product' && (yield client.query({
[dev:frontend] |                                             query: GET_RELATION_DATA,
[dev:frontend] |                                             variables: {
```

My code
```js
              query =
                value &&
                row.type === 'product' &&
                (await client.query({
                  query: GET_RELATION_DATA,
                  variables: {
                    model: 'product',
                    where: {
                      uid: value,
                    },
                    limit: 1,
                  },
                }));
```

**Comment by purplecandy** [comment]
Same error occurred 

```
./utils/index.ts
Module parse failed: The keyword 'yield' is reserved (104:28)
File was processed with these loaders:
 * ./node_modules/@next/react-refresh-utils/loader.js
 * ./node_modules/next/dist/build/webpack/loaders/next-swc-loader.js
You may need an additional loader to handle the result of these loaders.
|                                 })
|                             };
>                             yield client.request(mutations.INSERT_USER_QUESTION_ORDER, variables);
|                         }
```

starter happening when I using and async/await for promises in React components

```
"next": "^12.0.4",
```

**Comment by purplecandy** [comment]
Found a similar issue that might me useful
tldr: the issue only occurs in 12.0.4 the fix is already in canary we can use a 12.0.3 build
https://github.com/vercel/next.js/issues/31479

**Comment by balazsorban44** [comment]
Fixed in later releases. Just tested in `12.0.7`. Please upgrade.

For what it's worth, @felixmosh the following code is problematic in your example:

```jsx
  useEffect( async() => {
    await injectExternalScripts();
  },[])
```

See: https://reactjs.org/link/hooks-data-fetching

If it was only to showcase the issue, please ignore my comment.

**Comment by felixmosh** [comment]
@balazsorban44 thank you!

Why is the `useEffect` problematic? can you elaborate?

**Comment by Distortedlogic** [comment]
on vercel itself
![image](https://user-images.githubusercontent.com/54854447/146848092-7969aebe-7d6f-4b96-81cd-c5d764d9ee6f.png)

**Comment by Distortedlogic** [comment]
@felixmosh use this https://www.npmjs.com/package/use-async-effect

**Comment by Distortedlogic** [comment]
I solved my particular case. I was using await on a function that was no longer async, which threw the error

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
